### DISCUSSION TITLE
How to add chapters for a single "All-in-One"-mp3 File

### DISCUSSION THREAD

**Discussion by Gru0810** [advplyr/audiobookshelf#3092] [Q&A]
Sometimes I download mp3-Files from archive.org or convert it from youtube.
I always got one file. But sometimes I find informations about the Track like this:
![image](https://github.com/advplyr/audiobookshelf/assets/72630160/3f88dbfa-87ba-4970-90d8-7e6fd926836a)

How can/must I transform these information to add them to the OPF-File I use to define the Book?

**Comment by nichwall** [comment]
I'm not finding a standard on how chapters should be added to an OPF file for audiobooks.
https://idpf.org/epub/20/spec/OPF_2.0_final_spec.html

You can embed the metadata into the audio file, edit the chapter info using ABS, or edit `metadata.json` directly. If you're not embedding metadata, you will need to have some sort of converter script for other tools that might need the chapter information though.

**Comment by Gru0810** [comment]
I was able to add the chapter by using the builtIn-functionality. Works fine! Thanks!
